### PR TITLE
fix(identity): isolate in-memory roles by tenant

### LINKED ISSUE #8012: Identity in-memory roles are not safely isolated by tenant
## Problem

The current Identity in-memory role path can expose or overwrite roles across tenants.

## Evidence

- `RoleManager` creates deterministic kebab-case IDs but does not assign `TenantId`.
- `MemoryRoleStore` includes null-tenant roles for every tenant and keys its dictionary only by role ID.
- Two tenants creating the same role name can therefore address the same in-memory key.

Relevant source:

- `src/modules/Elsa.Identity/Services/RoleManager.cs`
- `src/modules/Elsa.Identity/Services/MemoryRoleStore.cs`

## Expected behavior

Roles listed, created, updated, and deleted in one tenant are isolated from every other tenant for both EF and in-memory stores.

## Context

Discovered while implementing elsa-workflows/elsa-studio#960. This issue is deliberately separate from the Studio feature; no silent Core scope expansion is planned there.

## Validation

Add cross-tenant contract tests covering duplicate role names and lifecycle isolation for the in-memory store.

### PR DESCRIPTION
**Pull request by sfmskywalker**
## Summary

- assign the ambient tenant when RoleManager creates a role, matching EF ApplyTenantId behavior on the in-memory path
- key MemoryRoleStore entries by tenant and role ID so same-name roles cannot overwrite each other
- scope updates/deletes to the tenant-filtered role records and retain legacy null-tenant records only for the default tenant
- add a lifecycle regression covering create/list/update/delete and duplicate names across tenants

Closes #8012

## Verification

- `dotnet test test/unit/Elsa.Identity.UnitTests/Elsa.Identity.UnitTests.csproj --no-restore --filter FullyQualifiedName~RoleManagerTests` (3 passed)
- `dotnet test test/unit/Elsa.Identity.UnitTests/Elsa.Identity.UnitTests.csproj --no-restore` (119 passed)
- `dotnet build src/modules/Elsa.Identity/Elsa.Identity.csproj --no-restore --configuration Release` (0 errors; 45 pre-existing XML documentation warnings across net8/net9/net10)

No `.codebase-memory` files were changed.

### CODE REVIEW

File: src/modules/Elsa.Identity/Services/MemoryRoleStore.cs [hunk 1]
```diff
@@ -62,13 +62,19 @@ public Task<IEnumerable<Role>> FindManyAsync(RoleFilter filter, CancellationToke
...
+    /// default in-memory stores had none at all. Null tenant IDs are retained only for the default tenant
+    /// for backwards compatibility with records created before tenant assignment was added.
     /// </remarks>
     private IQueryable<Role> Filter(IQueryable<Role> queryable, RoleFilter filter)
     {
         var tenantId = _tenantAccessor.TenantId;
-        queryable = queryable.Where(x => x.TenantId == tenantId || x.TenantId == Tenant.AgnosticTenantId || x.TenantId == null);
+        queryable = queryable.Where(x => x.TenantId == tenantId || x.TenantId == Tenant.AgnosticTenantId || (x.TenantId == null && tenantId == Tenant.DefaultTenantId));
 
         return filter.Apply(queryable);
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Identity/Services/MemoryRoleStore.cs:73]: This is already covered in the same change. RoleFilter.Apply includes null-tenant legacy records when TenantId is Tenant.DefaultTenantId, so the second filter preserves what the ambient filter admitted. The focused regression DefaultTenantListsLegacyRolesWithoutATenantId passes on head 54c7e31f6 (1/1), and the full Identity suite passed 120/120 before the main merge. No code change is needed.
